src/main/java/jp/co/soramitsu/sora/didresolver/commons/DIDTypeEnum.java

```diff
@@ -6,11 +6,11 @@
 
 @AllArgsConstructor
 public enum DIDTypeEnum {
-    IROHA("\\w*@\\w*\\..*"),
-    UUID("[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"),
-    ED("\\w*");
+  IROHA("(?<ACCOUNT>\\w+)@(?<DOMAIN>\\w+[\\w|.]*)"),
+  UUID("[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"),
```

Comment:
why do you change regexp?

Reply:
For the UUID: had to test if DID object was created properly, but couldn't - old regex did not match to UUID example in DID specification, I googled another regexp which is working fine

For the IROHA: had to capture specific part of pattern for later use
For the ED: \\w* was too ambiguous, it was matching an empty string which is incorrect